### DISCUSSION TITLE
how do i debug javascript using theia ide chrome / firefox?

### DISCUSSION THREAD

**Discussion by luiscipriano** [eclipse-theia/theia#9103] [Q&A]
Hi , I am setting up my final environment to program using theia ide, I generally work with wordpress. I already managed to get xdebug working on theia and several must-have extensions. But I still have a problem I can't debug javascript.

I am trying to debug javascript , using the following extensions:

Debugger for Chrome (I downloaded it from the microsoft market because the vsx registry gave me an error with this extension.)

JavaScript Debugger

and also test Debugger for Firefox

using the following settings ( launch.json )

{
      "name": "Launch chrome",
      "type": "chrome",
      "request": "launch",
      "url": "http: //localhost:80/",
      "webRoot": "${workspaceRoot}"
    },

I throw the following error:   Debug session initialization failed. See console for details.

    {
      "type": "chrome",
      "request": "attach",
      "name": "Attach chrome",
      "port": 9222
    },

I throw the following error: Debug session initialization failed. See console for details.

    {
            "name": "Launch firefox",
            "type": "firefox",
            "request": "launch",
            "reAttach": true,
            "file": "$ {workspaceFolder}/index.html"
        },

I throw the following error: Couldn't find the Firefox executable. Please specify the path by setting "firefoxExecutable" in your launch configuration.

    {
        "name": "Attach firefox",
            "type": "firefox",
            "request": "attach"
    }

i throw the following error : connect ECONNREFUSED 127.0.0.1:6000

I got these errors using the image I downloaded from docker hub " theiaide/theia "
and something very similar happens to me with the " theiaide/ theia-full "  image .

How can I make the javascript debug work?

Additional data:

My server from where I am doing this runs an ubuntu 20.04

**Comment by paul-marechal** [comment]
The chrome debugger works by opening a socket connection from where the extension runs to the Chrome process. From your logs it tried to connect to `127.0.0.1:6000`. You will most likely have difficulties getting your remote container connect to a local process.

**Reply by luiscipriano** [reply]
when i run docker i use the following command.

sudo docker run -it -p 3000: 3000 -p 8080: 8080 -p 9000: 9000 -p 80:80 -p 9222: 9222 -p 6000: 6000 --cap-add=NET_ADMIN -v "$(pwd):/home/luiss2cipriano/project:cached " theiaide/theia-full

Should that be enough to make port 6000 enabled to connect to the local process?

**Reply by paul-marechal** [reply]
Docker ports redirections are between the docker host and the container running on it. So if you connect to it from yet another machine then it won't be affected IIUC. Think of it like NAT rules. I don't even know if it would work when Chrome is running on the docker host, and I'd bet it doesn't work out of the box from a remote machine.

**Reply by paul-marechal** [reply]
https://docs.docker.com/engine/reference/commandline/run/#publish-or-expose-port--p---expose
